Lib/email/_header_value_parser.py

```diff
@@ -2013,12 +2013,15 @@ def get_address_list(value):
                 address_list.defects.append(errors.InvalidHeaderDefect(
                     "invalid address in address-list"))
         if value and value[0] != ',':
-            # Crap after address; treat it as an invalid mailbox.
-            # The mailbox info will still be available.
-            mailbox = address_list[-1][0]
-            mailbox.token_type = 'invalid-mailbox'
+            # Crap after address: if it's a single address then treat its
+            # mailbox as invalid. If it's a group, then just append the crap to
+            # it. The mailbox info will still be available.
+            dst = address_list[-1][0]
+            if dst.token_type != 'group':
+                dst.token_type = 'invalid-mailbox'
+
             token, value = get_invalid_mailbox(value, ',')
-            mailbox.extend(token)
+            dst.extend(token)
```

Comment:
Looking at this, I wonder if it would be better to be appending the invalid-mailbox at the level of the address list instead of appending it to the previous token (mailbox or group).  It's a semantic question: is the mailbox/group invalid because there's no trailing comma, or is the mailbox/group valid and what follows is an invalid mailbox?  I can see doing it either way, and it seems like there might be advantages to the latter.  The group case kind of highlights this because the semicolon is the end of the group, making the group valid.  The question is, would changing that for the invalid-mailbox case risk breaking anybody's code?

The problem with your solution here is that it doesn't mark the group as being invalid, which is then inconsistent with how a mailbox gets turned into an invalid-mailbox.  So it should either be invalid-group, or we should append the invalid-mailbox at the address-list level in both cases.  I'm favoring the latter currently unless someone sees a problem with that.

Note also that in your example it looks like that extra stuff is supposed to be a comment, which is just deprecated, not invalid (though I'd have to check the RFC to know if it is actually valid).  So maybe another problem is that we aren't recognizing the comment. Note that _header_value_parser doesn't really handle comments very well at all...it ignores them in some places, but that's about it.  My original plan included circling back to improve that, but it never happened.